addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,44 @@
+---Fields in module 'hr_timesheet'---
+hr_timesheet / account.analytic.account / company_uom_id (many2one)     : previously in module project
+hr_timesheet / account.analytic.account / project_count (integer)       : previously in module project
+hr_timesheet / account.analytic.account / project_ids (one2many)        : previously in module project
+hr_timesheet / project.project          / analytic_account_id (many2one): previously in module project
+hr_timesheet / project.task             / remaining_hours (float)       : previously in module project
+hr_timesheet / res.company              / project_time_mode_id (many2one): previously in module project
+# NOTHING TO DO: just moved fields, but nothing changes
+
+hr_timesheet / hr.employee              / currency_id (many2one)        : previously in module sale_timesheet
+hr_timesheet / hr.employee              / timesheet_cost (float)        : previously in module sale_timesheet
+# NOTHING TO DO: hr_timesheet is a dependency of sale_timesheet
+
+hr_timesheet / project.task             / children_hours (float)        : was renamed to subtask_effective_hours [nothing to do]
+# DONE: pre-migration: renamed field
+
+hr_timesheet / project.task             / delay_hours (float)           : DEL
+hr_timesheet / project.task             / total_hours (float)           : DEL
+# NOTHING TO DO
+
+hr_timesheet / res.company              / timesheet_encode_uom_id (many2one): NEW relation: uom.uom, required: required, req_default: function
+hr_timesheet / uom.uom                  / timesheet_widget (char)       : NEW 
+# NOTHING TO DO: new features
+
+---XML records in module 'hr_timesheet'---
+NEW ir.actions.act_window.view: hr_timesheet.timesheet_action_view_from_employee_form
+NEW ir.actions.act_window.view: hr_timesheet.timesheet_action_view_from_employee_list
+DEL ir.actions.act_window.view: hr_timesheet.timesheet_action_view_all_calendar
+DEL ir.actions.act_window.view: hr_timesheet.timesheet_action_view_calendar
+NEW ir.model.access: hr_timesheet.access_account_analytic_account_manager
+NEW ir.model.access: hr_timesheet.access_account_analytic_account_portal
+NEW ir.model.access: hr_timesheet.access_account_analytic_account_user
+NEW ir.model.access: hr_timesheet.access_account_analytic_line_project
+NEW ir.model.access: hr_timesheet.access_uom_uom_hr_timesheet
+DEL ir.model.access: hr_timesheet.access_product_product_user
+DEL ir.model.access: hr_timesheet.access_product_template_timesheet_user
+DEL ir.model.access: hr_timesheet.access_product_uom_hr_timesheet
+NEW ir.ui.view: hr_timesheet.account_analytic_account_view_form_inherit
+NEW ir.ui.view: hr_timesheet.assets_backend
+DEL ir.ui.view: hr_timesheet.timesheet_view_calendar
+NEW uom.uom: uom.product_uom_day (noupdate)
```

Comment:
These 2 updates are curious: they come from noupdate records in module `uom`, but here they are defined in XML as noupdate=0. On runbot, I have seen that the field is filled when you install `hr_timesheet`, no matter the original noupdate (I didn't know this). I suppose in this case it's the same, but I would like your confirmation, and this deserves at least a comment like:
```
NOTHING TO DO: Field `timesheet_widget` updated in noupdate records. Handled by standard update process.

Reply:
Can you confirm this?

Reply:
Confirmed, just by checking a v12 database.

Reply:
OK, thanks

Reply:
Wait. I checked now an old oca database migrated and the timesheet_widget was not filled. I will fully recheck properly.

Reply:
As it doesn't work, I put them in the noupdate_changes.xml